src/simulation/LoadGenerator.cpp

```diff
@@ -866,11 +1025,224 @@ LoadGenerator::manageOfferTransaction(
+        sorobanTransactionFrameFromOps(mApp.getNetworkID(), *account,
+                                       {uploadOp}, {}, uploadResources,
+                                       inclusionFee, resourceFee));
+    mPendingCodeKey = contractCodeLedgerKey;
+    return std::make_pair(account, tx);
+}
+
 std::pair<LoadGenerator::TestAccountPtr, TransactionFramePtr>
-LoadGenerator::sorobanTransaction(uint32_t numAccounts, uint32_t offset,
-                                  uint32_t ledgerNum, uint64_t accountId,
-                                  SorobanResources resources, size_t wasmSize,
-                                  uint32_t inclusionFee)
+LoadGenerator::createContractTransaction(uint32_t ledgerNum, uint64_t accountId,
+                                         uint32_t inclusionFee)
+{
+    releaseAssert(mCodeKey);
+    releaseAssert(mContractInstances.find(accountId) ==
+                  mContractInstances.end());
+
+    auto account = findAccount(accountId, ledgerNum);
+    SorobanResources createResources{};
+    createResources.instructions = 200'000;
+    createResources.readBytes = 5000;
+    createResources.writeBytes = 5000;
+
+    SCVal scContractSourceRefKey(SCValType::SCV_LEDGER_KEY_CONTRACT_INSTANCE);
+    auto salt = sha256(std::to_string(accountId));
+    auto [createOp, contractID] =
+        createSorobanCreateOp(mApp, createResources, *mCodeKey, *account,
+                              scContractSourceRefKey, salt);
+
+    auto const& instanceLk = createResources.footprint.readWrite.back();
+    ContractInstance instance;
+    instance.readOnlyKeys.emplace_back(*mCodeKey);
+    instance.readOnlyKeys.emplace_back(instanceLk);
+    instance.contractID = instanceLk.contractData().contract;
+    mContractInstances.emplace(accountId, instance);
+
+    auto resourceFee = sorobanResourceFee(mApp, createResources, 1000, 40);
+    resourceFee += 1'000'000;
+    auto tx = std::dynamic_pointer_cast<TransactionFrame>(
+        sorobanTransactionFrameFromOps(mApp.getNetworkID(), *account,
+                                       {createOp}, {}, createResources,
+                                       inclusionFee, resourceFee));
+
+    return std::make_pair(account, tx);
+}
+
+std::pair<LoadGenerator::TestAccountPtr, TransactionFramePtr>
+LoadGenerator::invokeSorobanLoadTransaction(uint32_t ledgerNum,
+                                            uint64_t accountId,
+                                            GeneratedLoadConfig const& cfg)
+{
+    auto account = findAccount(accountId, ledgerNum);
+    auto instanceIter = mContractInstances.find(accountId);
+    releaseAssert(instanceIter != mContractInstances.end());
+    auto const& instance = instanceIter->second;
+
+    auto const& networkCfg = mApp.getLedgerManager().getSorobanNetworkConfig();
+
+    // Approximate instruction measurements from loadgen contract. While the
+    // guest and host cycle counts are accurate, it very difficult to estimate
```

Comment:
I wouldn't spend much time on figuring this out, but these costs shouldn't be that difficult to figure out exactly, especially for the host calls. We do need to be vary of Wasm optimization for the guest cycles; but I'd hope we'll be able to freeze the loadgen contract soon. In any case, I think if cost per cycle has the correct upper bound, then we should never actually run out of resources at execution time. I suppose the tricky part is the storage cost, which depends on the inputs. 
So for now I would probably just rephrase this comment to explain where the cost discrepancies come from (IIUC currently these are just storage workload costs).